scipy/stats/tests/test_qmc.py

```diff
@@ -570,6 +591,20 @@ def test_warning(self):
             seed = np.random.RandomState(12345)
             qmc.LatinHypercube(1, seed=seed, optimization="toto")
 
+        with pytest.raises(ValueError, match=r"n is not the square of a"
+                                             r" prime number"):
```

Comment:
Just a habit I've gotten into for your consideration:
```suggestion
        message = r"n is not the square of a prime number"
        with pytest.raises(ValueError, match=message):
```
It's just a little bit nicer looking, and sometimes can save a few lines. When applied throughout a whole function, I think it makes a slight difference in readability.

Reply:
Agreed.